pkgs/development/libraries/openssl/default.nix

```diff
@@ -31,7 +31,12 @@ let
             --replace /bin/rm rm
         done
       '' + optionalString (versionAtLeast version "1.1.1") ''
-        substituteInPlace config --replace '/usr/bin/env' '${coreutils}/bin/env'
+        substituteInPlace config --replace '/usr/bin/env' '${
+          if stdenv.hostPlatform == stdenv.buildPlatform then
+            coreutils
+          else
+            buildPackages.coreutils
```

Comment:
Does `buildPackages.coreutils` work for both cases?

Reply:
I was worried about this line https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L403 where coreutils is being overriden and didn't realise it is actually the same thing. I will make correction.